spec/finalization-group.html

```diff
@@ -130,6 +130,9 @@ <h1>FinalizationGroup.prototype.register ( _target_ , _holdings_ [, _unregisterT
         1. If Type(_finalizationGroup_) is not Object, throw a *TypeError* exception.
         1. If _finalizationGroup_ does not have a [[Cells]] internal slot,
         throw a *TypeError* exception.
+        1. If Type(_unregisterToken_) is not Object,
+          1. If _unregisterToken_ is not *undefined*, through a *TypeError* exception.
```

Comment:
Should `null` be disallowed too?

Reply:
Is there precedent (JavaScript, not web) on whether `null` should be treated as `undefined` on an optional parameter, given that `undefined` is treated the same as omission of the optional parameter?

If there is clear precedent, and all clear precedent points in the same direction, we should do that.

Reply:
There unfortunately is not clear precedent - the majority of cases treat them the same (ie, ObjectCoercible or not), but almost as many cases treat them differently (default arguments in particular).

If the token is required to be an object, then i would expect one of two semantics: either null is the only way to indicate “nothing”, and undefined has a special default - or, null and undefined are treated the same, as a “nothing”.

Reply:
That's unfortunate. In that case, my preference is to treat `null` the same as `undefined`.

Reply:
I don't think it makes sense to try to treat `null` and `undefined` the same way in this case. Here, `undefined` indicates a missing argument, and so it's treated just like how argument defaults use undefined *only* for missing things. That's the case where there is no token.

I don't understand `@gsathya`'s original comment: `null` *is* disallowed and would trigger a TypeError, whereas `undefined` would be treated as the lack of an unregister token.

Reply:
Coming back to Mark's question, are there any cases that treat null as a missing argument? I don't think RequireObjectCoercible has to do with missing arguments.

Reply:
I just checked and isn't `Type(null)` the Null type?

Reply:
Array.from does, by way of https://tc39.github.io/ecma262/#sec-getmethod; both the Map and Set constructors treat null or undefined as a missing argument; to name a few.

Reply:
If we're to only allow object, disallowing `null` makes sense, which I guess is the case here since `Type(null)` is the Null type?

Reply:
I don't feel strongly about throwing an exception on null rather than treating it as a missing parameter, but I don't think either of these cases are about treating null as a missing parameter. They are about how null is treated as an iterable.

Reply:
Indeed - there’s not a reigning constraint. If the arg is an iterable, precedent would say to treat nullish the same. In this case, it seems like the intention is to allow only “an object” or “nothing”. Are there other APIs like that in the language we could look at, that take an optional argument, deny non-nullish primitives, and don’t ToObject?

Reply:
I still haven't see a case in JS where null is treated like a missing parameter (as opposed to a non-iterable value, or a non-object-coercible value). So I would say we do have a clear pattern to follow here.